surface/include/pcl/surface/impl/marching_cubes.hpp

```diff
@@ -116,25 +124,24 @@ pcl::MarchingCubes<PointNT>::createSurface (std::vector<float> &leaf_node,
   if (edgeTable[cubeindex] == 0)
     return;
 
-  //Eigen::Vector4f index_3df (index_3d[0], index_3d[1], index_3d[2], 0.0f);
-  Eigen::Vector3f center;// TODO coeff wise product = min_p_ + Eigen::Vector4f (1.0f/res_x_, 1.0f/res_y_, 1.0f/res_z_) * index_3df * (max_p_ - min_p_);
-  center[0] = min_p_[0] + (max_p_[0] - min_p_[0]) * float (index_3d[0]) / float (res_x_);
-  center[1] = min_p_[1] + (max_p_[1] - min_p_[1]) * float (index_3d[1]) / float (res_y_);
-  center[2] = min_p_[2] + (max_p_[2] - min_p_[2]) * float (index_3d[2]) / float (res_z_);
+  const Eigen::Vector3f size_voxel = (upper_boundary - lower_boundary).cwiseQuotient (
```

Comment:
https://github.com/t-lou/pcl/blob/10fe1eb3e7e720a4a80f8ae3c304a493eeb77785/surface/include/pcl/surface/impl/marching_cubes.hpp#L127-L128

is similar to this

https://github.com/t-lou/pcl/blob/10fe1eb3e7e720a4a80f8ae3c304a493eeb77785/surface/include/pcl/surface/impl/marching_cubes_hoppe.hpp#L66-L67

and this 

https://github.com/t-lou/pcl/blob/10fe1eb3e7e720a4a80f8ae3c304a493eeb77785/surface/include/pcl/surface/impl/marching_cubes_rbf.hpp#L103-L104

You're computing the same property three times. This normally hints me that it should become a member of the base class. What's your opinion?